src/components/data_submission/ConsentGroupForm.js

```diff
@@ -0,0 +1,445 @@
+      type: FormFieldTypes.MULTISELECT,
+      title: 'Data Location',
+      description: 'Please provide the location of your data resource for this consent group',
+      exclusiveValues: 'Not Determined',
+      options: dataLocationOptions,
+      placeholder: 'Data Location(s)',
+      validators: [FormValidators.REQUIRED],
+      defaultValue: consentGroup.dataLocation.map((loc) => {return {label: loc, value: loc};}),
+      onChange: ({key, value}) => onChange({key: key, value: value.map((sel) => sel.value)}),
+    }),
+
+    h(FormField, {
+      id: idx+'_url',
+      name: 'url',
+      title: 'Data URL',
+      validators: [FormValidators.REQUIRED, FormValidators.URL],
+      placeholder: 'Free text field for entering URL of data',
+      defaultValue: consentGroup.url,
+      onChange
+    }),
+  ]);
+};
+
+const ConsentGroupErrors = (props) => {
+
+  const {
+    errors
+  } = props;
+
+  return div({},
+    errors.map((err, idx) => {
+      return div({style: {marginBottom: '2rem'}}, [
+        h(Notification,
+          {
+            key: idx,
+            notificationData: {
+              level: 'danger',
+              message: err,
+            }
+          }),
+      ]);
+    }),
+  );
+};
+
+
+export const ConsentGroupForm = (props) => {
+  const {
+    idx,
+    saveConsentGroup,
+    deleteConsentGroup,
+  } = props;
+
+  const [consentGroup, setConsentGroup] = useState({
+    consentGroupName: '',
+
+    // primary: one of these needs to be filled
+    generalResearchUse: undefined,
+    hmb: undefined,
+    diseaseSpecificUse: undefined, // string
+    poa: undefined,
+    otherPrimary: undefined, // string
+
+    // secondary:
+    nmds: false, // No Methods Development or validation studies
+    gso: false, // genetic studies only
+    pub: false, // publication required
+    col: false, // collaboration required
+    irb: false, // irb approval required
+    gs: null, // string: geographic restriction
+    mor: false, // publication moratorium
+    npu: false, // non profit only
+    otherSecondary: null, // string
+
+    // dataLocation is one of:
+    // "AnVIL Workspace", "Terra Workspace",
+    // "TDR Location", "Not Determined"
+    dataLocation: [],
+
+    url: ''
+  });
+
+
+  const [consentGroupValidationErrors, setConsentGroupValidationErrors] = useState([]);
+  const [editMode, setEditMode] = useState(true);
+
+  return div({
+    style: {
+      border: '1px solid #283593',
+      padding: '1rem 2rem 1rem 2rem',
+      borderRadius: '4px',
+      marginBottom: '2rem',
+    },
+    id: idx+'_consentGroupForm'
+  }, [
+
+    h(ConsentGroupErrors,
+      {
+        errors: consentGroupValidationErrors,
+      }),
+
+    (editMode
+      ? h(EditConsentGroup, {
+        ...props,
+        ...{consentGroup: consentGroup, setConsentGroup: setConsentGroup},
+      })
+      : h(ConsentGroupSummary, {
+        ...props,
+        ...{consentGroup: consentGroup, id: idx+"_consentGroupSummary"},
```

Comment:
Codacy has a fix for the issue: [Strings must use singlequote.](https://app.codacy.com/gh/DataBiosphere/duos-ui/pullRequest?prid=9969451)
```suggestion
        ...{consentGroup: consentGroup, id: idx+'_consentGroupSummary'},
```